packages/react-server/src/ReactFlightServer.js

```diff
@@ -174,6 +200,30 @@ function attemptResolveElement(
       case REACT_MEMO_TYPE: {
         return attemptResolveElement(type.type, key, ref, props);
       }
+      case REACT_PROVIDER_TYPE: {
+        pushProvider(type._context, props.value);
+        if (__DEV__) {
+          const extraKeys = Object.keys(props).filter(value => {
+            if (value === 'children' || value === 'value') {
+              return false;
+            }
+            return true;
+          });
+          if (extraKeys.length !== 0) {
+            console.error(
+              'ServerContext can only have a value prop and children. Found: %s',
+              JSON.stringify(extraKeys),
+            );
+          }
+        }
+        return [
+          REACT_PROVIDER_TYPE,
+          type._context._globalName,
+          key,
+          // Rely on __popProvider being serialized last to pop the provider.
+          {...props, __popProvider$$: type._context},
```

Comment:
Instead of spreading, you can just pass `value` and `children` which is a little faster than assign.

```
{
value: props.value,
children: props.children,
__pop: type._context
}
```

That way you're also sure you don't have any conflicts. If the props contained a key with that name then it wouldn't be put at the end if it already existed and then it wouldn't pop at the right place. Potentially a security issue.